dion/__init__.py

```diff
@@ -0,0 +1,8 @@
+from .dion import Dion
+from .dion import DionMixedPrecisionConfig
+from .dion import DionParamConfig
```

Comment:
`DionParamConfig` and `DionReferenceParamConfig` are purely for internal use and should not be available for pip package users to import. They shouldn't be in `__init__` and you can rename them to `_DionParamConfig` in their respective files.